pkg/filesystem/subvolume.go

```diff
@@ -131,6 +139,47 @@ func getK8sRefSubvolume(ctx context.Context, clientsets *k8sutil.Clientsets) map
 	return subvolumeNames
 }
 
+// getk8sRefSnapshotHandle returns the snapshothandle for k8s ref of the volume snapshots
+func getK8sRefSnapshotHandle(ctx context.Context, clientsets *k8sutil.Clientsets) map[string]snapshotInfo {
+
+	snapConfig, err := snapclient.NewForConfig(clientsets.KubeConfig)
+	if err != nil {
+		logging.Fatal(err)
+	}
+	snapList, err := snapConfig.VolumeSnapshotContents().List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Error fetching volumesnapshotcontents: %v\n", err))
+	}
+
+	snapshotHandles := make(map[string]snapshotInfo)
+	for _, snap := range snapList.Items {
+		if snap.Status != nil && snap.Status.SnapshotHandle != nil {
+
+			snapshotHandleId := getSnapshotHandleId(*snap.Status.SnapshotHandle)
+			// map the snapshotHandle id to later lookup for the subvol id and
+			// match the subvolume snapshot.
+			snapshotHandles[snapshotHandleId] = snapshotInfo{}
+		}
+	}
+
+	return snapshotHandles
+}
+
+// getSnapshotHandleId gets the id from snapshothandle
+// SnapshotHandle: 0001-0009-rook-ceph-0000000000000001-17b95621-
+// 58e8-4676-bc6a-39e928f19d23
+// SnapshotHandleId: 17b95621-58e8-4676-bc6a-39e928f19d23
+func getSnapshotHandleId(snapshotHandle string) string {
+	// get the snaps id from snapshot handle
+	splitSnapshotHandle := strings.SplitAfterN(snapshotHandle, "-", 6)
+	if len(splitSnapshotHandle) < 6 {
+		return ""
+	}
+	snapshotHandleId := splitSnapshotHandle[len(splitSnapshotHandle)-1]
```

Comment:
nit: return it directly instead of declaring a var
```suggestion
	return splitSnapshotHandle[len(splitSnapshotHandle)-1]
```